src/main/java/frameworksdrivers/Gateways/StudentDbGateway.java

```diff
@@ -155,11 +131,13 @@ private void saveUserNoCourses(CreateStudentDsModel student){
     @Override
     public void saveGPA(EvaluatorDbRequestModel requestModel) {
         try {
-            String SQL = "UPDATE " + DATABASE_NAME + " SET CourseGrade = ? WHERE StudentID = ? AND CourseID = ?";
+            String SQL = "INSERT INTO " + DATABASE_NAME + " (StudentID, Password, CourseID, CourseGrade) " +
```

Comment:
You're right with the idea of creating a new row but my one issue is that if the student takes a course again the course is already a row for them so shouldn't you check for the course within the student  similar to what we do in the enrollment use case and then based on whether that is true or not either update an already existing row or insert a  new one?